src/main/java/entities/NoValidCardException.java

```diff
@@ -0,0 +1,7 @@
+package entities;
+
+public class NoValidCardException extends Exception{
```

Comment:
Is this class still necessary? If we are considering changing the return of selectRandomCard from a potential error to null.

Reply:
It would not be no, it has been deleted in the latest push.